test/integration/work_test.go

```diff
@@ -554,4 +554,218 @@ var _ = ginkgo.Describe("ManifestWork", func() {
 			util.AssertAppliedManifestWorkDeleted(appliedManifestWork.Name, spokeWorkClient, eventuallyTimeout, eventuallyInterval)
 		})
 	})
+
+	ginkgo.Context("Resource sharing and adoption between manifestworks", func() {
+		var anotherWork *workapiv1.ManifestWork
+		var anotherAppliedManifestWorkName string
+		ginkgo.BeforeEach(func() {
+			manifests = []workapiv1.Manifest{
+				util.ToManifest(util.NewConfigmap(o.SpokeClusterName, "cm1", map[string]string{"a": "b"}, []string{})),
+				util.ToManifest(util.NewConfigmap(o.SpokeClusterName, "cm2", map[string]string{"c": "d"}, []string{})),
+			}
+		})
+
+		ginkgo.JustBeforeEach(func() {
+			// Create another manifestworks with one shared resource.
+			// TODO We might not want the sharing in this cases, since the content of a resource in two manifestworks
+			// can be different.
+			anotherWork = util.NewManifestWork(o.SpokeClusterName, "sharing-resource-work", []workapiv1.Manifest{manifests[0]})
+			anotherWork, err = hubWorkClient.WorkV1().ManifestWorks(o.SpokeClusterName).Create(context.Background(), anotherWork, metav1.CreateOptions{})
+			anotherAppliedManifestWorkName = fmt.Sprintf("%s-%s", hubHash, anotherWork.Name)
+			gomega.Expect(err).ToNot(gomega.HaveOccurred())
+		})
+
+		ginkgo.It("shared resource between the manifestwork should be recreated when one manifestwork is deleted", func() {
```

Comment:
after the orphaning policy is done, this test should be updated to ensure nothing is deleted.  I suspect that's where you'll hit issues